slither/tools/kspec_coverage/analysis.py

```diff
@@ -0,0 +1,215 @@
+import re
+from math import sqrt
+from slither.core.solidity_types.user_defined_type import UserDefinedType
+from slither.solc_parsing.declarations.function import Function
+from slither.core.declarations.solidity_variables import SolidityFunction
+from tabulate import tabulate
+
+def get_all_covered_kspec_functions(target):
+    # Create a set of our discovered functions which are covered
+    covered_functions = set()
+
+    BEHAVIOUR_PATTERN = re.compile('behaviour\s+(\S+)\s+of\s+(\S+)')
+    INTERFACE_PATTERN = re.compile('interface\s+([^\r\n]+)')
+
+    def refactor_type(type):

```

Comment:
Is there a particular reason to have an inner function?

I would recommend creating a private function instead. Overall inner functions are slower to execute and make the code more difficult to maintain.